lms/tasks/email_digests.py

```diff
@@ -119,28 +119,60 @@ def send_instructor_email_digest_tasks(*, batch_size):  # pylint:disable=unused-
     rate_limit="10/m",
 )
 def send_instructor_email_digest(
-    *, h_userid: str, created_after: str, created_before: str, **kwargs
+    *, h_userid: str, created_before: str, **kwargs
 ) -> None:
     """
     Generate and send instructor email digests to the given users.
 
-    The email digests will cover activity that occurred in the time period
-    described by the `created_after` and `created_before` arguments.
+    The email digests will cover activity that occurred up until
+    `created_before`.
 
     :param h_userid: the h_userid of the instructor to email
-    :param created_after: the beginning of the time period as an ISO 8601 format string
-    :param created_before: the end of the time period as an ISO 8601 format string
+    :param created_before: cut-off time after which activity will not be
+        included in the email, as an ISO 8601 format string
     :param kwargs: other keyword arguments to pass to DigestService.send_instructor_email_digest()
     """
     # Caution: datetime.fromisoformat() doesn't support all ISO 8601 strings!
     # This only works for the subset of ISO 8601 produced by datetime.isoformat().
-    created_after = datetime.fromisoformat(created_after)
     created_before = datetime.fromisoformat(created_before)
 
     with app.request_context() as request:  # pylint:disable=no-member
         with request.tm:
+            task_done_data = _get_task_done_data(request.db, h_userid)
+
+            created_after = created_before - timedelta(days=7)
+
+            if task_done_data:
+                created_after = max(
+                    datetime.fromisoformat(task_done_data["created_before"]),
+                    created_after,
+                )
+
             digest_service = request.find_service(DigestService)
 
             digest_service.send_instructor_email_digest(
                 h_userid, created_after, created_before, **kwargs
```

Comment:
`DigestService.send_instructor_email_digest()` is unchanged: it still takes both `created_after` and `created_before` arguments. This is good because it means we don't need to make any changes to `DigestService`. Also, the email admin views call `DigestService` with hard-coded `created_after` and `created_before` arguments and this means that we don't need to change those admin views either. This is especially nice because data has been pre-created in the QA and production DBs so that the admin pages hard-coded `created_after` and `created_before` match some annotations, so that easy way of end-to-end testing the email feature will still work without changes.